app/analysis/index/controller.js

```diff
@@ -462,4 +460,69 @@ export default class AnalysisController extends Controller.extend(
       Object.keys(this.allQueryParams).filter((k) => k !== "ordering")
     );
   }
+
+  get allQueryParams() {
+    const queries = {};
+    for (const param of this.queryParams) {
+      // the type of param is string of object that have one key
+      if (typeof param === "string") {
+        queries[param] = this[param];
+        continue;
+      }
+      const key = Object.keys(param)[0];
+      queries[key] = this[key];
+    }
+    return queries;
+  }
+
+  get queryParamsState() {
+    const states = {};
+    for (const param of this.queryParams) {
+      // the type of param is string of object that have one key
```

Comment:
Could you clarify this comment? What is that *special case* you are handling there with the `Object.keys(param)[0]`?

Reply:
is string `OR` object that have one key

Reply:
I could imagine that ~of~ **or** typo. But as a reader of your code I want to know why we have to do this *special case* handling here. What query param requires this and why?

Reply:
there is something wrong that i understand from the documentation, i thought that default query parmas value should be there